bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java

```diff
@@ -242,17 +250,37 @@ public BigtableSession(BigtableOptions opts) throws IOException {
 
     clientInterceptors =
         clientInterceptorsList.toArray(new ClientInterceptor[clientInterceptorsList.size()]);
+    // END set up interceptors
 
+    // BEGIN set up Data Clients
+    // This is needed for the throttling client / Bulk Mutation which is only cloud-bigtable-client.
     Channel dataChannel = getDataChannelPool();
 
     BigtableSessionSharedThreadPools sharedPools = BigtableSessionSharedThreadPools.getInstance();
 
-    // More often than not, users want the dataClient. Create a new one in the constructor.
     CallOptionsFactory.ConfiguredCallOptionsFactory callOptionsFactory =
         new CallOptionsFactory.ConfiguredCallOptionsFactory(options.getCallOptionsConfig());
-    dataClient =
-        new BigtableDataGrpcClient(dataChannel, sharedPools.getRetryExecutor(), options);
-    dataClient.setCallOptionsFactory(callOptionsFactory);
+
+    if (options.useGCJClient()) {
+      BigtableDataSettings dataSettings =
+          BigtableVeneerSettingsFactory.createBigtableDataSettings(options);
+      this.dataGCJClient = new BigtableDataGCJClient(
```

Comment:
Why not merge dataGCJClient and dataClient into a single member variable of type IBigtableDataClient?

Reply:
We still need to have access to the underlying `dataClient` directly for the users that use it directly.  Eventually, we may be able to deprecate the getter, but for now, we'll have to keep it around.

Reply:
with this pr that getter will return null? If so, I think it would be better to use IBigtableDataClient, and in getter check if the underlying client is the classic client and return it. If its gcj to throw a NotImplemented exception

Reply:
+1 on cleanup.  NotImplement exception will cause our integration tests to fail.  Would a warning be enough for now?

Reply:
sure

Reply:
Can we do the refactoring now (i.e. just move the creation of GCJ into the constructor), and do the other cleanup in another PR?

Reply:
ok